cowboys/crud.py

```diff
@@ -0,0 +1,36 @@
+from sqlalchemy.orm import Session
+from sqlalchemy import func
+
+from . import models, schemas
+
+def crear_cowboy (db: Session, cowboy: schemas.CrearCowboy):
+    db_cowboy = models.Cowboy (     name = cowboy.name,
+                                    email = cowboy.email,
+                                    tonto = cowboy.tonto if cowboy.tonto is not None else 0
+                                    )
+    db.add (db_cowboy)
+    # guarda la nueva instansia de cowboy
+    db.commit()
+    # Actualiza el objeto db_cowboy con los valores de la base de datos
+    # db.refresh (db_cowboy)
+    return db_cowboy
+
+# optiene un usuario por mail
+def cowboy_existente (db: Session, cowboy ):
+    if db.query (models.Cowboy).filter (models.Cowboy.email == cowboy.email).first():
+        resultado = "email"
+        return resultado
+
+    return True
```

Comment:
Esta funcion esta un poco confusa.
Porque a veces retorna `True` y otras veces retorna `email`.


Cuando el cowboy existe se retorna el valor `email`. 
Y cuando el cowboy no existe se retorna `True`. Esto es contra intuitivo.



Una funcion llamada `cowboy_existente` que retorna `True` cuando el cowboy no existe es como guardarse una pistola cargada y sin seguro en el bolsillo del pantalon...

Mi sugerencia es que la funcion retorne `True` cuando el cowboy existe y `False` cuando el cowboy no existe.

Te dejo dos sugerencias abajo.

Reply:
Sugerencia 1.

```suggestion
def cowboy_existente (db: Session, cowboy ):
    if db.query (models.Cowboy).filter (models.Cowboy.email == cowboy.email).first():
        return True

    return False
```

Reply:
Sugerencia 2.

Usando `.one_or_none()` del ORM de FastAPI que retorna un solo valor si exist o `None` si no existe.

```suggestion
def cowboy_existente (db: Session, cowboy ):
    return db.query (models.Cowboy).filter (models.Cowboy.email == cowboy.email).one_or_none() is not None
```